### PR TITLE
Change `filter` -> `filters` in configuration highlights

### LINKED ISSUE #140: Sensu "filter" vs. "filters" attributes
Doc reference/handlers.html 

It was recently recommended/suggested that some improvements be made to the event filtering documentation: 

- make sure you specify in enterprise docs that one can still use custom filters next to built-in ones
- please add a clear example config where the custom filter is applied; it took me a while to figure out where exactly to apply it

### PR DESCRIPTION
**Pull request by Blue0ctober**
## Description
Change use of `filter` to `filters` to encourage users to use the plural form. 

## Motivation and Context
This is to help future proof use of `filters` allowing us to eventually decommission the use of `filter`. This also helps prevent a potential issue of a user having both a `filter` and `filters` configured in a handler and then not having the `filter` be applied since `filters` will override `filter`. 

Closes  #140

## How Has This Been Tested?
tested in local hugo dev

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New doc/guide
- [x] Fixing errata

## Checklist:
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] All tests have passed.

### CODE REVIEW

**Comment by Blue0ctober** [comment]
I need someone to review 2.0 doc changes and make sure they look appropriate.

File: content/sensu-core/0.29/guides/intro-to-filters.md [hunk 1]
```diff
@@ -98,6 +100,9 @@ in the handler definition. For example:
 
 To specify multiple Sensu event filters, use the `filters` attribute (plural).
 
+_WARNING: `filter` attribute will be depreciated in future releases. Please use
```
Comment (cwjohnston) [inline content/sensu-core/0.29/guides/intro-to-filters.md:103]: @portertech per https://docs.sensu.io/sensu-core/2.0/reference/handlers/#handler-attributes it looks like `filter` singular form is not supported in 2.0. Can you confirm?

File: content/sensu-core/0.29/reference/handlers.md [hunk 2]
```diff
@@ -186,7 +186,7 @@ Handler set definitions allow groups of handlers (i.e. individual collections of
 actions to take on event data) to be referenced via a single named handler set.
 
 _NOTE: Attributes defined on handler sets do not apply to the handlers they
-include. For example, `filter`, `filters`, and `mutator` attributes defined
```
Comment (cwjohnston) [inline content/sensu-core/0.29/reference/handlers.md]: I don't think we'll be removing `filter` from Sensu Core 1.x, so let's keep `filter` mentions in the handler reference.

File: content/sensu-core/1.0/reference/handlers.md [hunk 3]
```diff
@@ -185,8 +185,11 @@ events.
 Handler set definitions allow groups of handlers (i.e. individual collections of
```
Comment (cwjohnston) [inline content/sensu-core/1.0/reference/handlers.md:185]: keep filter mentions in the handler reference.

File: content/sensu-core/1.1/reference/handlers.md [hunk 4]
```diff
@@ -186,7 +186,7 @@ Handler set definitions allow groups of handlers (i.e. individual collections of
 actions to take on event data) to be referenced via a single named handler set.
 
 _NOTE: Attributes defined on handler sets do not apply to the handlers they
-include. For example, `filter`, `filters`, and `mutator` attributes defined
```
Comment (cwjohnston) [inline content/sensu-core/1.1/reference/handlers.md]: keep filter mentions in the handler reference.

File: content/sensu-core/1.3/reference/handlers.md [hunk 5]
```diff
@@ -186,7 +186,7 @@ Handler set definitions allow groups of handlers (i.e. individual collections of
 actions to take on event data) to be referenced via a single named handler set.
 
 _NOTE: Attributes defined on handler sets do not apply to the handlers they
-include. For example, `filter`, `filters`, and `mutator` attributes defined
```
Comment (cwjohnston) [inline content/sensu-core/1.3/reference/handlers.md]: keep filter mentions in the handler reference.

File: content/sensu-core/1.2/reference/handlers.md [hunk 6]
```diff
@@ -186,7 +186,7 @@ Handler set definitions allow groups of handlers (i.e. individual collections of
 actions to take on event data) to be referenced via a single named handler set.
 
 _NOTE: Attributes defined on handler sets do not apply to the handlers they
-include. For example, `filter`, `filters`, and `mutator` attributes defined
```
Comment (cwjohnston) [inline content/sensu-core/1.2/reference/handlers.md]: keep filter mentions in the handler reference.

File: content/sensu-core/2.0/reference/filters.md [hunk 7]
```diff
@@ -20,10 +20,10 @@ filter analysis flow performs these steps:
 
 * When the Sensu server is processing an event, it will check for the definition
 of a `handler` (or `handlers`). Prior to executing each handler, the Sensu
-server will first apply any configured `filter` (or `filters`) for the handler.
+server will first apply any configured `filters` for the handler.
 * If multiple `filters` are configured for a handler, they are executed
 sequentially.
-* Filter `statements` are compared with event data.
```
Comment (cwjohnston) [inline content/sensu-core/2.0/reference/filters.md:26]: I don't think this line should change, it's describing individual named filters.

File: content/sensu-core/2.0/reference/filters.md [hunk 8]
```diff
@@ -20,10 +20,10 @@ filter analysis flow performs these steps:
 
 * When the Sensu server is processing an event, it will check for the definition
 of a `handler` (or `handlers`). Prior to executing each handler, the Sensu
-server will first apply any configured `filter` (or `filters`) for the handler.
```
Comment (cwjohnston) [inline content/sensu-core/2.0/reference/filters.md:23]: Need confirmation that `filter` singular is removed in 2.0

File: content/sensu-core/2.0/reference/handlers.md [hunk 9]
```diff
@@ -51,7 +51,7 @@ of actions to take on event data) to be referenced via a single named handler
 set.
 
 _NOTE: Attributes defined on handler sets do not apply to the handlers they
-include. For example, `filter`, `filters`, and `mutator` attributes defined
```
Comment (cwjohnston) [inline content/sensu-core/2.0/reference/handlers.md:54]: Need confirmation that filter singular is removed in 2.0

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-core/0.29/guides/intro-to-filters.md (hunk 1 above)
Comment (cwjohnston) [reply content/sensu-core/0.29/guides/intro-to-filters.md:103]: Sean has confirmed that the singular form `filter` is removed in 2.0 so this deprecation warning is 👌

File: content/sensu-core/2.0/reference/filters.md (hunk 8 above)
Comment (cwjohnston) [reply content/sensu-core/2.0/reference/filters.md:23]: Confirmed 👍

**Comment by Blue0ctober** [comment]
@cwjohnston we still want to remove the following from `handlers.md`?

```
filter       | 
-------------|------
description  | The Sensu event filter (name) to use when filtering events for the handler.
required     | false
type         | String
example      | {{< highlight shell >}}"filter": "occurrences"{{< /highlight >}}
```

**Comment by cwjohnston** [comment]
@blue0ctober no, I don’t think it should be removed.  We should leave the singular form in the handler reference as it is still supported.

**Comment by Blue0ctober** [comment]
Closing in favor of #463

**Pull request closed without merging**